### PR TITLE
Extended documentation on Subscriptions for v12 (#5295)

### PR DESCRIPTION
**Pull request by sunghwan2789**
Co-authored-by: Lukáš Procházka <lukasino97@gmail.com>
(cherry picked from commit 60b8957114556b105cba36aa96128b41e3e882ea)

Cherry-picked #5295  to make it live.

### CODE REVIEW

**Comment by sunghwan2789** [comment]
I have signed CLA, but @CLAassistant don't update the comment!

**Comment by michaelstaib** [comment]
Hey @sunghwan2789 can you sign the CLA please?

**Comment by sunghwan2789** [comment]
Oh @cts-tradeit has to sign the CLA..
Is this happening because the commit is cherry-picked?

**Comment by cts-tradeit** [comment]
@sunghwan2789 Ive checked, cla was signed under this acc aswell

File: website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md [hunk 1]
```diff
@@ -252,6 +252,8 @@ public async Book PublishBook(Book book, [Service] ITopicEventSender sender)
 
 If more complex topics are required, we can use the `ITopicEventReceiver`.
 
+> ⚠️ Note: Pay attention to the use of `SubscribeAndResolve` instead of `Subscribe`. As can be seen in the code-first example above, all subscription fields are in fact made of two methods (resolver and subscriber). When using `Subscribe`, the attributed method represents resolver method (with subscriber method generated at runtime) whereas `SubscribeAndResolve` denotes subscriber method (with resolver method generated at runtime).
```
Comment (michaelstaib) [inline website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: This is a bit unclear... I would recommend people not to use SubscribeAndResolve... we are deprecating it. Instead I would use the two separate methods. Subscribe and resolve leads to issues for the standard user with captured contexts.

Comment (michaelstaib) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: I think the intend for the note is to dissuade people from using SubscribeAndResolve... this should be clearer.

Comment (michaelstaib) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: In the V12 documentation it should be still available ... but for 13 we can remove it

Comment (michaelstaib) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: > We recommend not use `SubscribeAndResolve` as it very easily can lead to unexpected behavior with the state that is aggregated on not released.

For 12 we could have a very strong first sentence and for 13 remove it from the docs.

@tobias-tengler what do you think?

Comment (tobias-tengler) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: I think we can remove it from the v13 documentation. We've already documented that it's deprecated in the upgrade guide so I think we should stop documenting it as well.

Comment (michaelstaib) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md:255]: @sunghwan2789 can you apply the changes so we can merge?

Comment (sunghwan2789) [reply website/src/docs/hotchocolate/v12/defining-a-schema/subscriptions.md]: I applied the changes. Thanks!

**Pull request closed without merging**